File: BackEnd/Application/UseCases/Documents/UpdateDocument/UpdateDocumentCommandHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using Infrastructure.UnitOfWork;
using MediatR;

namespace Application.UseCases.Documents.UpdateDocument;

public class UpdateDocumentCommandHandler(IRepository<Document, Guid> documentRepository, IUnitOfWork unitOfWork) : IRequestHandler<UpdateDocumentCommand, bool>
{
    public async Task<bool> Handle(UpdateDocumentCommand request, CancellationToken cancellationToken)
    {
        var document = await documentRepository.GetByIdAsync(request.Id, cancellationToken);
        if (document == null) return false;
        document.Update(request.ObjectName);
        await documentRepository.UpdateAsync(document);
        await unitOfWork.SaveChangesAsync(cancellationToken);
        return true;
    }
}

File: flutter/lib/application/use_cases/documents/update_document/update_document_command_handler.dart
import 'package:enterprise_management/application/use_cases/documents/update_document/update_document_command.dart';
import 'package:enterprise_management/infrastructure/repositories/document_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/command_handler.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class UpdateDocumentCommandHandler extends ICommandHandler<UpdateDocumentCommand, bool> {
  const UpdateDocumentCommandHandler({required this._documentRepository});
  final IDocumentRepository _documentRepository;

  @override
  Future<Result<bool>> handle(UpdateDocumentCommand command) async {
    final res = await _documentRepository.updateDocument(id: command.id, objectName: command.objectName);
    return Result.success(res);
  }

}